### PR TITLE
Add specs

### PR DESCRIPTION
**Pull request by dce**
Modify middleware so that it uses response's `.each` instead of `.body`, as per Rack specification.

### CODE REVIEW

**Comment by dce** [comment]
cc @efatsi

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+source 'https://rubygems.org'
+
+# Specify your gem's dependencies in colonel_kurtz_ruby.gemspec
```
Comment (ltk) [inline Gemfile]: `olive_branch.gemspec`?

Comment (dce) [inline Gemfile]: Yesssss

**Comment by ltk** [comment]
hella tite :+1:

File: lib/olive_branch/middleware.rb [hunk 2]
```diff
@@ -14,18 +14,20 @@ def call(env)
...
+          if inflection == "camel"
+            new_response.deep_transform_keys! { |k| k.camelize(:lower) }
+          elsif inflection == "dash"
+            new_response.deep_transform_keys!(&:dasherize)
+          end
 
-        [status, headers, [new_response.to_json]]
-      else
-        [status, headers, response]
+          body.replace(new_response.to_json)
```
Comment (efatsi) [inline lib/olive_branch/middleware.rb]: where'd you find this?

**Comment by efatsi** [comment]
tests look good

File: lib/olive_branch/middleware.rb (hunk 2 above)
Comment (dce) [inline lib/olive_branch/middleware.rb]: `.replace` is just a thing you can do to strings: http://ruby-doc.org/core-1.9.3/String.html#method-i-replace

Never had much use for it before now, but I needed some way to manipulate the response, and `ActionDispatch::Response::RackBody` doesn't give you many (any?) other options.

According to the Rack website (http://rack.github.io/), the response body "must respond to `each`", so this seemed like the most universal approach.

**Comment by dce** [comment]
Thanks dudes.

**Pull request closed without merging**